presto-main/src/main/java/io/prestosql/server/TaskResource.java

```diff
@@ -110,9 +111,17 @@ public TaskResource(
     @ResourceSecurity(INTERNAL_ONLY)
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
+    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo, @QueryParam("state") String stateFilter)
```

Comment:
Adding possibility to filter out specifically on state field feels very arbitrary to me. Why not do the filtering on the caller side according to specific needs?

Reply:
> Adding possibility to filter out specifically on state field feels very arbitrary to me. Why not do the filtering on the caller side according to specific needs?

We actually want to get running taskinfo just like get running queryInfo in QueryResource.java, but if there is no filtering state, we will get a lot of finished taskinfo.

Reply:
I understand. But I do no necessarily see why is that a problem. Should be easy to filter out those even in shell using `jq`. And in proper scripting language even easier.

Reply:
Yes, this is not a bug, it is an improvement. With this feature, my upgrade script does not need to download dozens of workers every few seconds, each worker 40-50MB taskinfo when checking whether the worker still has a running task.

Reply:
Thanks for explaining motivation @shenh062326. I am leaning towards merging this one. @electrum can you chime in and shout if you feel that such an extension to protocol something you are very much against?